src/GogeToken.sol

```diff
@@ -1251,30 +1185,44 @@ contract DogeGaySon is ERC20, Ownable {
         );
     }
 
-    function swapAndSendCakeDividends(uint256 tokens) private {
+    function swapAndSendCakeDividends(uint256 tokens) internal {
         swapTokensForDividendToken(tokens, address(this), cakeDividendToken);
         uint256 cakeDividends = IERC20(cakeDividendToken).balanceOf(address(this));
-        transferDividends(cakeDividendToken, address(cakeDividendTracker), cakeDividendTracker, cakeDividends);
+
+        assert(IERC20(cakeDividendToken).transfer(address(cakeDividendTracker), cakeDividends));
+
+        cakeDividendTracker.distributeDividends(cakeDividends);
+        emit SendDividends(cakeDividends);
     }
     
-    function transferToWallet(address payable recipient, uint256 amount) private {
+    function transferToWallet(address payable recipient, uint256 amount) internal {
         recipient.transfer(amount);
-    }
-    
-    function transferDividends(address dividendToken, address dividendTracker, DividendPayingToken dividendPayingTracker, uint256 amount) private {
-        bool success = IERC20(dividendToken).transfer(dividendTracker, amount);
-        
-        if (success) {
-            dividendPayingTracker.distributeDividends(amount);
-            emit SendDividends(amount);
-        }
+        emit RoyaltiesTransferred(recipient, amount);
     }
     
     function _transferOwnership(address newOwner) external {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "Not authorized");
         require(newOwner != address(0));
+
         super.transferOwnership(newOwner);
+        
+        isExcludedFromFees[newOwner] = true;
+        cakeDividendTracker.excludeFromDividends(newOwner);
     }
+
+    /// @notice Withdraw a gogeToken from the treasury.
+    /// @dev    Only callable by owner.
+    /// @param  _token The token to withdraw from the treasury.
+    function safeWithdraw(address _token) external onlyOwner {
+        uint256 amount = IERC20(_token).balanceOf(address(this));
+        require(amount > 0, "GogeToken.sol::safeWithdraw() IERC20(_token).balanceOf(address(this)) == 0");
```

Comment:
`Insufficient token balance` would be more concise for the `require` message lol.

Reply:
Fixed